sklearn/datasets/_samples_generator.py

```diff
@@ -1411,6 +1411,22 @@ def make_low_rank_matrix(
     -------
     X : ndarray of shape (n_samples, n_features)
         The matrix.
+
+    Examples
+    --------
+    >>> from numpy.linalg import svd
+    >>> from sklearn.datasets import make_low_rank_matrix
+    >>> X = make_low_rank_matrix(n_samples=50,
+    ...     n_features=25,
+    ...     effective_rank=5,
+    ...     tail_strength=0.01,
+    ...     random_state=0,
+    ... )
+    >>> X.shape
+    (50, 25)
+    >>> U, s, V = svd(X)
+    >>> s.sum()
+    5...
```

Comment:
I don't think that this is a valid confirmation of the effective rank. The rank is not the sum of the singular values but the number of non-zero singular values. Suppose X has rank 5, then 10*X has also rank 5 but the sum of its singular values is 10 times bigger.

Reply:
Good point. Our test is therefore quite bad 😁

Reply:
Apparently the profile is not the one that we expect. The singular value should follow a exponential decay controlled by the effective rank parameter. So we should probably avoid this last line.